pkg/compress/compress.go

```diff
@@ -0,0 +1,42 @@
+package compress
+
+import (
+	"fmt"
+	"sync"
+)
+
+var (
+	compressions    = map[string]Compression{}
+	compressionsMut sync.Mutex
+)
+
+type Compression interface {
+	Compress([]byte) ([]byte, error)
+	Decompress([]byte) ([]byte, error)
+}
+
+// RegisterCompression registers a custom compression algorithm.
+func RegisterCompression(name string, c Compression) error {
+	if c == nil {
+		return nil
+	}
+	compressionsMut.Lock()
+	defer compressionsMut.Unlock()
+	_, ok := compressions[name]
+	if ok {
```

Comment:
Returning an error when `ok` is `true` is disturbing me. Maybe you could rename to `found` or `exists`?